bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/discovery/HDPowerViewSddpDiscoveryParticipant.java

```diff
@@ -56,18 +66,26 @@ public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
         final ThingUID thingUID = getThingUID(device);
         if (thingUID != null) {
             try {
+                String label;
+                String serial;
                 int generation = getGeneration(device);
-                String label = generation == 3 //
-                        ? String.format("@text/%s [\"%s\"]", LABEL_KEY_GATEWAY, device.ipAddress)
-                        : String.format("@text/%s [\"%s\", \"%s\"]",
-                                HDPowerViewHubMDNSDiscoveryParticipant.LABEL_KEY_HUB, generation, device.ipAddress);
-
+                if (generation < 3) {
+                    label = String.format("@text/%s [\"%s\", \"%s\"]",
+                            HDPowerViewHubMDNSDiscoveryParticipant.LABEL_KEY_HUB, generation, device.ipAddress);
+                    serial = propertyGetter.getSerialNumberApiV1(device.ipAddress);
+                } else {
+                    label = String.format("@text/%s [\"%s\"]", LABEL_KEY_GATEWAY, device.ipAddress);
+                    serial = propertyGetter.getSerialNumberApiV3(device.ipAddress);
+                }
                 DiscoveryResult hub = DiscoveryResultBuilder.create(thingUID)
                         .withProperty(HDPowerViewHubConfiguration.HOST, device.ipAddress)
-                        .withRepresentationProperty(HDPowerViewHubConfiguration.HOST).withLabel(label).build();
+                        .withProperty(Thing.PROPERTY_SERIAL_NUMBER, serial)
+                        .withRepresentationProperty(Thing.PROPERTY_SERIAL_NUMBER).withLabel(label).build();
                 logger.debug("SDDP discovered Gen {} hub/gateway '{}' on host '{}'", generation, thingUID,
                         device.ipAddress);
                 return hub;
+            } catch (HubException e) {
+                logger.debug("Error discovering hub", e);
```

Comment:
Really minor, but if you did this and removed the two lines below:
```suggestion
            } catch (IllegalArgumentException | HubException e) {
                logger.debug("Error discovering hub", e);
```

You could replace lines 124-126 by:
```java
        throw new IllegalArgumentException("Device has unexpected 'type' property");
```

since we would have centralized the logging.